### ISSUE TITLE
3.18.0 asks for "all files" permission, even if previously declined

### ISSUE THREAD

**Issue by silsha** [nextcloud/android#9327]
### Steps to reproduce
1. Make sure Nextcloud doesn't have access permission to "all files"
2. Open Nextcloud 3.18.0 (you will get a dialog to accept the permissions)
3. Confirm the Dialog with "OK", don't accept the permissions in the next view
4. You can now use the Nextcloud app without issues
5. Close the nextcloud app and reopen it
6. Go to 3. and repeat

### Expected behaviour
- Nextcloud should honour if the user does not want to grant access to all files and should not ask for access every time the user starts Nextcloud.

### Actual behaviour
- Nextcloud asks for the access permissions on every launch

### Environment data

Nextcloud app version: 3.18.0

Nextcloud server version: 22.2.3

Reverse proxy: No

**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by Shrizt** [comment]
I suggest to be able to choose which files I'd like to grant for NC
and it's for sure not all files on device
issue confirmed

**Comment by chrfreitag** [comment]
Same here after the upgrade to 3.18.0. I did not grant the permission on startup after the upgrade. Then I noticed that I could not use the upload activity of the app anymore as I was used to because of the missing permission. Apparently I had granted the permission in the past. The OS seems to have forgotten about it on the upgrade to version 3.18.0.

My comment might not help solve the issue on the reoccuring prompt on permission denial, but It might give a reason for why the prompt is occuring in the first place for users having upgraded from a previous version.

**Comment by AndyScherzinger** [comment]
> Would scoped storage be an option here as Shrizt perhaps had in mind?

No, this is the exact reason on why this pops-up now. 3.18 is targeting sdk30 which then leads to scoped storage. With up/down/auto-up-load and being a file app you basically need (and are qualified) to request this permission. And we need it as you can see for various reasons. Being able to request this permission means you had to be validated for it by Google itself, so we when through the application process for this permission with Google and it was granted.

So
* we need they permission
* not granting it renders parts of the app disfunctional

**Comment by silsha** [comment]
@AndyScherzinger 
I understand the technical reasoning, but from the user's point of view this will not be acceptable.

I am reporting the problem for one of our customers. Their employees use the Nextcloud app exclusively to retrieve certain documents remotely.

They were skeptical as to why the app suddenly needed access to all files, completely justifiedly so in my opinion
Obviously, the app works for this purpose even if you don't give the permissions.

My suggestion would therefore be to only request the permissions when they are really needed and point out the possible consequences (which parts of the app no longer work?).
Alternatively, you should at least clearly explain to the user why these permissions are now suddenly needed.

**Comment by AlvaroBrey** [comment]
@silsha Perhaps a better explanation is in order, but we don't really request any more access than before*. The Files app already requested `READ_EXTERNAL_STORAGE` before, which was the pre-android10 equivalent. The UX for requesting files access has just changed to intentionally make it more scary on Google's part.

That being said, there is a valid point in using the application in a "only-read-from-server" way, so it may be a good idea to try to implement that. However this is not an easy feat in Android 10+, as many features such as auto-upload will be permanently broken, and others will have to be tweaked to work differently without the permission. As @AndyScherzinger said, our hands are sort-of tied now because of Android 10 policies.

* if anything, we request less permissions now. On Android 10+ we don't have access to the `Android/data` directory on the external storage with this new schema.

tl;dr:
* We aren't actually asking for more access than before, although it's true that it isn't optional anymore.
* Perhaps a single dialog is not enough to explain why the UX for data access has changed
* We'll try to think of a way to balance the needs of people who want Nextcloud to access all files and of other people who want Nextcloud to have the least amount of access possible.

**Comment by AndyScherzinger** [comment]
@silsha so basically what @AlvaroBrey said giving some more details about the change, so let's see how we can further improve it.
